tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,338 @@
+    assert_script_run("virsh net-create vnet_routed.xml");
+    save_screenshot;
+    upload_logs "vnet_routed.xml";
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "ROUTED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_routed --live");
+        script_retry "journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
+        my $gi_vnet_routed = script_output("journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        #Confirm ROUTED NETWORK
+        #TODO
+        #There should be added the static routes to LAN router, but we have not permission
+        #to modify LAN router.So, before confirm routed network setting, confirm current
+        #routed netwrok WITHOUT testing ping from guest to external
+        assert_script_run("ssh root\@$gi_vnet_routed 'ip a'", 60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy ROUTED NETWORK
+    assert_script_run("virsh net-destroy vnet_routed");
+    save_screenshot;
+
+##ISOLATED NETWORK
+##Prepare Test
+    my $vnet_isolated = <<'EOF';
+<network>
+ <name>vnet_isolated</name>
+ <bridge name="virbr2"/>
+ <ip address="192.168.152.1" netmask="255.255.255.0">
+  <dhcp>
+   <range start="192.168.152.2" end="192.168.152.254" />
+  </dhcp>
+ </ip>
+</network>
+EOF
+
+    #Create ISOLATED NETWORK
+    assert_script_run("echo '$vnet_isolated' > vnet_isolated.xml");
+    assert_script_run("virsh net-create vnet_isolated.xml");
+    save_screenshot;
+    upload_logs "vnet_isolated.xml";
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "ISOLATED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_isolated --live");
+        script_retry "journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
+        my $gi_vnet_isolated = script_output("journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        #Confirm ISOLATED NETWORK
+        assert_script_run("! ssh root\@$gi_vnet_isolated 'ping -c2 -W1 openqa.suse.de'");
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy ISOLATED NETWORK
+    assert_script_run("virsh net-destroy vnet_isolated");
+    save_screenshot;
+
+    #Restore Guest systems
+    restore_guests;
+
+    #Restore br123 for virt_autotest
+    restore_standalone;
+
+    #Restore Network setting
+    restore_network;
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+
+    #Restart libvirtd service
+    restart_libvirtd;
+
+    #Restore Guest systems
+    restore_guests;
+
+    #Destroy created virtual networks
+    destroy_vir_network;
+
+    #Restore br123 for virt_autotest
+    restore_standalone;
```

Comment:
Would it be better if restore_standalone happens before restore_guests ?

Reply:
Great, had updated code as restore_standalone happens before restore_guests.

Reply:
Done